### ISSUE TITLE
refactor(architecture): Clarify agent vs. strategy entry decision authority

### ISSUE THREAD

**Issue by kimrejstrom** [kimrejstrom/alpacalyzer-algo-trader#79]
## Feature Description

**HIGH PRIORITY**: The agent → strategy → execution decision flow is confusing and inconsistent. Agents provide trade setup (entry, stop, target, size), but strategies may reject or re-calculate values. This needs to be clarified per user decision: "Agents decide, strategies must be clear what strategy is the trigger."

### Current State (Confusing)

**Decision chain:**
```
Risk Manager → Portfolio Manager → Trading Strategist (GPT-4)
    ↓ provides TradingStrategy object
    ↓ contains: entry_point, stop_loss, target_price, quantity, trade_type
    ↓
ExecutionEngine.evaluate_entry()
    ↓ calls Strategy.evaluate_entry()
    ↓
MomentumStrategy/BreakoutStrategy/MeanReversionStrategy
    ↓ CURRENTLY: Re-validates technical conditions
    ↓ MAY: Reject agent's setup
    ↓ MAY: Re-calculate stop/target (MomentumStrategy:700-701)
```

### Problems

1. **Double Validation**: Agents (GPT-4) + strategy both validate entry
2. **Unclear Authority**: Who has final say? Can strategy override agent?
3. **Redundant Calculations**: MomentumStrategy calculates its own stop/target in some paths
4. **Inconsistent Behavior**: Different strategies handle agent recommendations differently

### Intended Behavior (Per User Decision)

**"Agents decide, strategies must be clear what strategy is the trigger"**

This means:

**Agents' Responsibility:**
- Analyze ticker from multiple perspectives (technical, sentiment, value)
- Generate optimal trade setup: entry point, stop loss, target, quantity
- Provide trade_type (long/short) and reasoning
- **NO technical filtering** - let strategies do that

**Strategies' Responsibility:**
- Validate that the setup fits their specific trading philosophy
- **ONLY reject** if conditions don't match their style
- **NEVER** re-calculate entry, stop, target, or quantity
- Use agent values directly

**Examples of Valid Rejections:**

| Strategy | Would Reject | Reason |
|-----------|---------------|---------|
| MomentumStrategy | Negative momentum entry | "Momentum requires positive trend" |
| BreakoutStrategy | No consolidation pattern | "No clear consolidation to break out of" |
| MeanReversionStrategy | RSI not oversold | "Not in mean reversion range" |

**Examples of Invalid Rejections (Don't do these):**
- "I think price is wrong" → Use agent's entry_point
- "My stop is tighter" → Use agent's stop_loss
- "I want more/less shares" → Use agent's quantity

---

## Technical Requirements

- [ ] Document the agent → strategy decision flow in code comments
- [ ] Update all strategies to consistently accept agent recommendations
- [ ] Remove redundant calculations from MomentumStrategy (lines 700-701)
- [ ] Add tests verifying agent values are used unchanged
- [ ] Update AGENTS.md with clarified authority flow
- [ ] Update README.md if needed

---

### Task 1: Document decision flow in strategies

**Files:**
- Modify: `src/alpacalyzer/strategies/base.py` (add docstring to Strategy protocol)
- Modify: `src/alpacalyzer/strategies/momentum.py`
- Modify: `src/alpacalyzer/strategies/breakout.py`
- Modify: `src/alpacalyzer/strategies/mean_reversion.py`

**Step 1: Update Strategy protocol docstring**

Read `src/alpacalyzer/strategies/base.py`, find the Strategy protocol class or BaseStrategy class.

Add to docstring:
```python
class BaseStrategy(ABC):
    """
    Base class for trading strategies.
    
    DECISION FLOW:
    
    1. Agent (GPT-4 via TradingStrategist) provides:
       - entry_point: Suggested entry price
       - stop_loss: Suggested stop loss price
       - target_price: Suggested take profit price
       - quantity: Suggested number of shares
       - trade_type: 'long' or 'short'
    
    2. Strategy validates setup against its philosophy:
       - MomentumStrategy: Checks positive momentum trend
       - BreakoutStrategy: Checks for consolidation pattern
       - MeanReversionStrategy: Checks oversold conditions
    
    3. If conditions don't match, STRATEGY REJECTS:
       - Return EntryDecision(should_enter=False, reason="...")
       - Agent's setup is discarded
    
    4. If conditions match, STRATEGY USES AGENT VALUES:
       - Use agent's entry_point, stop_loss, target_price, quantity
       - DO NOT recalculate these values
       - Return EntryDecision(should_enter=True, ...)
    
    AUTHORITY:
    - Agents have authority to propose trade setups
    - Strategies have authority to reject if setup doesn't fit their style
    - Strategies MUST NOT override agent's calculated values
    """
```

**Step 2: Add comment to MomentumStrategy**

At `src/alpacalyzer/strategies/momentum.py:106-115`, add:
```python
# Use agent recommendation values directly for trade setup
# Strategy's role: Validate technical conditions, not recalculate prices
# Agent provides: entry_point, stop_loss, target_price, quantity
# Strategy validates: Momentum trend, technical signals
return EntryDecision(
    should_enter=True,
    reason=f"Entry conditions met ({conditions_met_count}/{total_conditions}), trade_type={trade_type}",
    suggested_size=agent_recommendation.quantity,  # Use agent's quantity
    entry_price=agent_recommendation.entry_point,  # Use agent's entry
    stop_loss=agent_recommendation.stop_loss,  # Use agent's stop
    target=agent_recommendation.target_price,  # Use agent's target
)
```

**Step 3: Add comment to BreakoutStrategy**

At `src/alpacalyzer/strategies/breakout.py:131`, add:
```python
"""
Breakout Strategy Entry Decision Flow:

NOTE: BreakoutStrategy does not use agent_recommendation parameter.
Unlike MomentumStrategy, BreakoutStrategy detects opportunities independently
through technical analysis (consolidation + breakout pattern).

If agent_recommendation is provided in future:
- BreakoutStrategy would validate consolidation condition exists
- Use agent's entry/stop/target/quantity values
- Reject if no consolidation pattern detected
"""
```

**Step 4: Add comment to MeanReversionStrategy**

Similar to MomentumStrategy, document the flow.

**Step 5: Commit documentation**
```bash
git add src/alpacalyzer/strategies/
git commit -m "docs(strategies): Document agent vs. strategy decision flow for #79"
```

---

### Task 2: Remove redundant calculations from MomentumStrategy

**Files:**
- Modify: `src/alpacalyzer/strategies/momentum.py`

**Step 1: Identify redundant code**

Read `src/alpacalyzer/strategies/momentum.py` lines 700-701:
```python
# Lines 700-701 - REDUNDANT (already using agent values above)
stop_loss = price - (atr * 1.5)  # ❌ REDUNDANT
target = price + (atr * 4.5)  # ❌ REDUNDANT
```

These are calculated but **never used** because lines 108-114 already use agent values.

**Step 2: Remove redundant code**

Delete lines 700-701 (verify exact line numbers).

**Step 3: Run existing tests**
```bash
uv run pytest tests/strategies/test_momentum.py -v
```

Expected: PASS (no behavior change, just cleanup)

**Step 4: Commit cleanup**
```bash
git add src/alpacalyzer/strategies/momentum.py
git commit -m "refactor(strategies): Remove redundant calculations from MomentumStrategy for #79"
```

---

### Task 3: Add test verifying agent values are used

**Files:**
- Test: `tests/strategies/test_momentum.py` (add new test)
- Test: `tests/strategies/test_breakout.py` (add new test)

**Step 1: Add test for MomentumStrategy**

```python
def test_momentum_strategy_uses_agent_values():
    """Test that MomentumStrategy uses agent recommendation unchanged."""
    from alpacalyzer.strategies.momentum import MomentumStrategy
    from alpacalyzer.strategies.base import MarketContext
    from alpacalyzer.data.models import TradingStrategy
    from alpacalyzer.analysis.technical_analysis import TradingSignals
    
    # Setup
    strategy = MomentumStrategy()
    context = MarketContext(
        vix=20.0,
        market_status="open",
        account_equity=100000.0,
        buying_power=50000.0,
        existing_positions=[],
        cooldown_tickers=[]
    )
    
    # Create mock signal
    signal = create_mock_signal("AAPL")
    
    # Create agent recommendation
    agent_rec = TradingStrategy(
        ticker="AAPL",
        quantity=100,
        entry_point=150.0,
        stop_loss=145.5,
        target_price=163.5,
        risk_reward_ratio=3.0,
        strategy_notes="Test",
        trade_type="long",
        entry_criteria=[]
    )
    
    # Evaluate
    decision = strategy.evaluate_entry(signal, context, agent_rec)
    
    # Verify agent values are used unchanged
    assert decision.suggested_size == 100, f"Should use agent quantity, got {decision.suggested_size}"
    assert decision.entry_price == 150.0, f"Should use agent entry, got {decision.entry_price}"
    assert decision.stop_loss == 145.5, f"Should use agent stop, got {decision.stop_loss}"
    assert decision.target == 163.5, f"Should use agent target, got {decision.target}"
```

**Step 2: Run test**
```bash
uv run pytest tests/strategies/test_momentum.py::test_momentum_strategy_uses_agent_values -v
```

Expected: PASS

**Step 3: Commit**
```bash
git add tests/strategies/test_momentum.py
git commit -m "test(strategies): Add test for agent value usage in MomentumStrategy for #79"
```

---

### Task 4: Update AGENTS.md documentation

**Files:**
- Modify: `AGENTS.md`

**Step 1: Add section on decision flow**

Find appropriate location in `AGENTS.md` (perhaps in "Architecture Overview" or create new section).

Add:
```markdown
### Agent vs. Strategy Decision Authority

The trading system uses a two-tier decision model:

**Tier 1: Agent (GPT-4 via TradingStrategist)**
- Analyzes ticker from multiple perspectives
- Proposes optimal trade setup:
  * Entry point (price)
  * Stop loss
  * Take profit target
  * Position size (quantity)
  * Trade direction (long/short)
- Provides reasoning

**Tier 2: Trading Strategy**
- Validates that agent's setup fits strategy's philosophy
- Can REJECT if conditions don't match:
  * MomentumStrategy: Rejects negative momentum, weak technicals
  * BreakoutStrategy: Rejects no consolidation pattern
  * MeanReversionStrategy: Rejects not-oversold conditions
- MUST USE agent values for entry/stop/target/size (no recalculation)

**Key Principle**
> Agents propose, strategies validate. Strategies never override agent's calculated trade parameters.

**Example Valid Flow:**
1. Agent: "Buy AAPL @ $150, stop $145.5, target $163.5, 100 shares"
2. MomentumStrategy: Validates momentum positive, technicals strong ✓
3. ExecutionEngine: Submits order using agent's values

**Example Valid Rejection:**
1. Agent: "Buy AAPL @ $150, stop $145.5, target $163.5, 100 shares"
2. MomentumStrategy: "Momentum -5% (negative), rejects" ✗
3. ExecutionEngine: Discards signal

**What Strategies Don't Do:**
- Don't recalculate entry price
- Don't change stop loss level
- Don't modify take profit target
- Don't adjust quantity based on their own logic
```

**Step 2: Commit documentation**
```bash
git add AGENTS.md
git commit -m "docs(agents): Clarify agent vs. strategy decision authority for #79"
```

---

### Task 5: Update README.md if needed

**Files:**
- Modify: `README.md` (only if architecture section needs update)

**Step 1: Review architecture section**

Check if README.md has agent/strategy flow description. If so, update to reflect clarified authority.

**Step 2: Commit**
```bash
git add README.md
git commit -m "docs(readme): Update architecture section with decision flow for #79"
```

---

### Task 6: Run full test suite and linting

**Step 1: Run strategy tests**
```bash
uv run pytest tests/strategies/ -v
```

**Step 2: Run execution tests**
```bash
uv run pytest tests/test_execution/ -v
```

**Step 3: Run linting**
```bash
uv run ruff check src/alpacalyzer/strategies/
uv run ruff format src/alpacalyzer/strategies/
uv run ty check src/alpacalyzer/strategies/
```

**Step 4: Run full test suite**
```bash
uv run pytest tests -v > test-output.txt 2>&1
# Check for regressions
```

**Step 5: Final commit**
```bash
git add .
git commit -m "chore(strategies): Run linting and type checking for #79"
```

---

## Acceptance Criteria

- [ ] Given agent recommendation, When MomentumStrategy evaluates entry, Then agent's entry/stop/target/quantity are used unchanged
- [ ] Given agent recommendation, When Strategy rejects, Then reason clearly states why (based on strategy philosophy, not disagreement with prices)
- [ ] Given MomentumStrategy code, When reviewed, Then redundant calculations (lines 700-701) are removed
- [ ] Given AGENTS.md, When read, Then agent vs. strategy authority is clearly documented
- [ ] All tests pass
- [ ] Code is linted and documented

---

## Out of Scope

- Changing strategy algorithms (only clarifying decision flow)
- Modifying how TradingStrategist generates recommendations
- Changing agent decision logic
- Implementing new rejection criteria (just documenting existing ones)

---

## Documentation to Check

- `src/alpacalyzer/strategies/base.py` - BaseStrategy docstring
- `src/alpacalyzer/strategies/momentum.py` - Redundant code lines 700-701
- `src/alpacalyzer/trading/trading_strategist.py` - Agent generates TradingStrategy
- `AGENTS.md` - Development guidelines
- `README.md` - Architecture overview
- `HARDENING_PLAN.md` - This issue's parent plan

---

### Why This Is Confusing

The system evolved over time:
1. Originally: Strategies calculated everything themselves
2. Then: Agents added to provide trade setup
3. Migration: New structure but old patterns remained

Now we have:
- Strategies that still think they control prices
- Agents that provide complete setups
- No clear authority boundary

### Testing Approach

- **Verify behavior**: Test that agent values are passed through unchanged
- **Test rejections**: Verify strategies can reject with clear reasons
- **Document flow**: Make authority explicit in code and docs

### Future Considerations

If we want strategies to be able to adjust agent values in the future:
1. This issue would need to be revisited
2. New decision model would be: Agent proposes, Strategy can tweak
3. For now: Agent proposes, Strategy only validates/rejects

---

**Issue Reference**: #79
**Parent Plan**: HARDENING_PLAN.md
**Estimated Time**: 2-3 hours
**Priority**: P0-High
